docs/feature-flags.md

```diff
@@ -0,0 +1,31 @@
+# Feature flags and partial releases
+
+Feature flags are an important tool that enables [trunk-based development](https://trunkbaseddevelopment.com/). They allow in-progress features to be merged into the main branch while still allowing that branch to be deployed to production at any time, thus decoupling application deploys from feature releases. For a deeper introduction, [Martin Fowler's article on Feature Toggles](https://martinfowler.com/articles/feature-toggles.html) and [LaunchDarkly's blog post on feature flags](https://launchdarkly.com/blog/what-are-feature-flags/) are both great articles that explain the what and why of feature flags.
+
+## How it works
+
+This project leverages [Amazon CloudWatch Evidently](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Evidently.html) to create and manage feature flags.
+
+## Creating feature flags
+
+The list of feature flags for an application is defined in the `feature_flags` property in its app-config module (in `/infra/[app_name]/app-config/main.tf`). To create a new feature flag, add a new string to that list. To remove a feature flag, remove the feature flag from the list. The set of feature flags will be updated on the next terraform apply of the service layer, or during the next deploy.
+
+## Querying feature flags in the application
+
+To determine whether a particular feature should be enabled or disabled for a given user, the application code calls an "is feature enabled" function in the feature flags module. Under the hood, the module will call AWS Evidently's [EvaluateFeature](https://docs.aws.amazon.com/cloudwatchevidently/latest/APIReference/API_EvaluateFeature.html) API to determine whether a feature is enabled or disabled. For partial rollouts, it will remember which variation of the application a particular user saw and keep the user experience consistent for that user. For more information about the feature flags module, look in the application code and docs.
+
+## Managing feature releases and partial rollouts via AWS Console
+
+The system is designed to allow the managing of feature releases and partial rollouts outside of terraform, which empowers business owners and product managers to control
+
+### To enable or disable a feature
```

Comment:
I selfishly wanted to avoid putting screenshots since screenshots are likely to become out of date when AWS can change their GUI / design so it increases the maintenance burden. Is there anything with words that I can use, like maybe add some text in bold not to change the variation names or values? If you feel strongly about it I'm happy to put a screenshot, but just sharing my preference.

<img width="812" alt="feature-variations" src="https://github.com/navapbc/template-infra/assets/447859/85510e99-fc0d-4890-98fb-25e8054a2a6d">

Reply:
I'll leave the image out for now but willing to add it later in a later doc update